Update for SauceLabs without using the outdated Maven plugin (#1545)
vaadin/docs: articles/testing/running-test-on-multiple-browsers.asciidoc

[[testbench.grid.sauce-labs]]
== Using Sauce Labs Test Grid
To run your tests on Sauce Labs you first need to get an account at https://saucelabs.com/ and take note of your user name and access key.

> Before starting a test, you need to setup the https://docs.saucelabs.com/secure-connections/sauce-connect/quickstart/[Sauce Connect proxy]. The proxy creates a secure connection between the Sauce Labs browser environment and your machine so the Sauce Labs browsers can access a server you are running locally. You should run the proxy with a tunnel identifier (`-i some-tunnel-identifier`) to avoid conflicts with other tunnels you or a CI server might create later.

Reviewer: ```suggestion
Before starting a test, you need to set up the https://docs.saucelabs.com/secure-connections/sauce-connect/quickstart/[Sauce Connect proxy].
The proxy creates a secure connection between the Sauce Labs browser environment and your machine, so the Sauce Labs browsers can access a server you are running locally.
You should run the proxy with a tunnel identifier (`-i some-tunnel-identifier`) to avoid conflicts with other tunnels that you or a CI server might create later.
```

> When the proxy is up and running, you can run your tests by passing the Sauce Labs username and access key to it. This can be done using system properties, e.g.

Reviewer: ```suggestion
When the proxy is up and running, you can run your tests by passing the Sauce Labs username and access key to it.
This can be done using system properties, e.g.
```

```sh
mvn verify -Dsauce.user=<yourusername> -Dsauce.sauceAccessKey=<youraccesskey> -Dsauce.tunnelId=some-tunnel-identifier
```
or by first setting the credentials as environment variables and then running the tests, e.g.
```sh
export SAUCE_USERNAME=<yourusername>
export SAUCE_ACCESS_KEY=<youraccesskey>
export SAUCE_TUNNEL_ID=some-tunnel-identifier
mvn verify
```

> When writing your tests there is nothing special about Sauce Labs that needs to be taken into account. You write them as described in <<testbench.grid.test-remote>>

Reviewer: ```suggestion
When writing your tests, there is nothing special about Sauce Labs that needs to be taken into account.
You write them as described in <<testbench.grid.test-remote>>
```

> When running tests on a CI/build server you typically have a build step before the tests where the Sauce Connect Proxy is started and a build step after the tests where the tunnel is teared down.

Reviewer: ```suggestion
When running tests on a CI/build server, you typically have a build step before the tests, where the Sauce Connect Proxy is started, and a build step after the tests, where the tunnel is torn down.
```
